### ISSUE TITLE
Newbie questions

### ISSUE THREAD

**Issue by christiansakai** [davezuko/react-redux-starter-kit#545]
Hello, a newcomer here, been trying to use React/Redux for a while now but keep cowering due to being overwhelmed by too many build tools.  Finally settled on this boilerplate! It's clean! have all the features that I need and pretty easy to follow through! thanks @davezuko and the contributors!

I've few questions, mostly opinionated:
1. How do I make the tests files go in each modules/views/components instead in one tests folder that this boilerplate right now
2. I am trying to use this approach [Domain Driven Design](http://marmelab.com/blog/2015/12/17/react-directory-structure.html) to structure my React/Redux app, is there anything specific I should change?
3. What is jsconfig.json used for?
4. What is NODE_ENV=staging for in travis.yml?
5. When I tried to run `npm run test` it doesn't have 100% coverage, is this ok? Sorry I'm pretty newbie in this code coverage things.
6. Seems like in many other boilerplates, Koa are beginning to replace Express as the standard, however  following koa's repo, seems like Koa 2 alpha is not ready yet and many middleware probably need to catch up, is this a concern? If I were to downgrade to Koa 1, is there anything specific needs to be changed?
7. How do I keep my forked repo in sync with what changed here? Should I just use the rebase upstream approach?

Sorry for asking a lot! thanks for the reply before!

**Comment by davezuko** [comment]
Will write more answers as I have time.

1) https://github.com/davezuko/react-redux-starter-kit/blob/master/tests/test-bundler.js#L23 This line of code is responsible for importing the tests, simply point it to your source directory. You'll also then want to update the regex below to ignore `*.spec.js` files, otherwise those will be included in your coverage report.

2) To come

3) This is for [vscode](https://code.visualstudio.com/Docs/languages/javascript), it's not relevant to any other IDE.

4) Staging is just there as a test to make sure the application can be built with a `NODE_ENV` that is not development/production, and `staging` is a common environment name. It's basically there to show that you could have staging-specific configuration if you so wanted (think of changing the target API endpoints between dev/staging/production).

5) Yes, that's expected. I should probably get around to implementing full code coverage, though.

6) To come

7) To come. We actually use a modified version of this boilerplate at my work, and have an update utility that will sync all apps based on the template to include the latest changes, but it's not yet open sourced. If that changes, I will definitely let you know.

Will probably flesh these out a bit more, but those are just some quick responses before I head out for the night.

**Comment by christiansakai** [comment]
@davezuko thank you so much for the responses! much appreciated. I could submit a PR for helping implementing the code coverage.

**Comment by davezuko** [comment]
If you'd like to that would be much appreciated. Going to close this now since it's not technically an issue, but we can continue the discussion still.

**Comment by christiansakai** [comment]
@davezuko thanks, yeah. I'll be waiting on more explanation, meanwhile trying to navigate through the boilerplate

**Comment by christiansakai** [comment]
@davezuko When I run npm run test, 

```
ERROR LOG: 'Warning: Failed propType: Required prop `counter` was not specified in `HomeView`.'
ERROR LOG: 'Warning: Failed propType: Required prop `doubleAsync` was not specified in `HomeView`.'
ERROR LOG: 'Warning: Failed propType: Required prop `increment` was not specified in `HomeView`.'
```

this shows up because the props weren't put in the tests, which is okay, but seems bothering for the eyes, lol. Is there any way to fix this?

One way I can think about is to put it outside the class definition

```
if (process.env.NODE_ENV !== 'test') {
  HomeView.propTypes = {
    counter: PropTypes.number.isRequired,
    doubleAsync: PropTypes.func.isRequired,
    increment: PropTypes.func.isRequired
  };
}
```

**Comment by davezuko** [comment]
I should just fix the tests ;)

**Comment by christiansakai** [comment]
lok okay :+1:

**Comment by christiansakai** [comment]
Now when I run `npm run test`, it stuck until this log below, doesn't go to the tests.  I feel that I didn't change anything, do you have any idea what might have happened?

```
⇒  npm run test:dev

> react-redux-starter-kit@1.0.0 test:dev /Users/christiansakai/Desktop/fullstack-react-redux-starter-kit
> npm run test -- --watch


> react-redux-starter-kit@1.0.0 test /Users/christiansakai/Desktop/fullstack-react-redux-starter-kit
> better-npm-run test "--watch"

running better-npm-run in /Users/christiansakai/Desktop/fullstack-react-redux-starter-kit
Executing script: test

to be executed: babel-node ./node_modules/karma/bin/karma start bin/karma --watch
  app:config Create configuration. +0ms
  app:config Apply environment overrides for NODE_ENV "test". +1ms
  app:config No configuration overrides found for NODE_ENV "test" +1ms
  app:webpack:config Create configuration. +793ms
  app:webpack:config Apply ExtractTextPlugin to CSS loaders. +6ms
  app:karma Create configuration. +1ms
12 02 2016 18:02:07.157:WARN [karma]: No captured browser, open http://localhost:9876/
12 02 2016 18:02:07.167:INFO [karma]: Karma v0.13.20 server started at http://localhost:9876/
12 02 2016 18:02:07.178:INFO [launcher]: Starting browser PhantomJS
12 02 2016 18:02:07.967:INFO [PhantomJS 2.1.1 (Mac OS X 0.0.0)]: Connected on socket /#lTk-oMqPbKiReReXAAAA with id 66887604

```

**Comment by davezuko** [comment]
Downgrade and lock karam to `0.13.19`, there are some bugs with the latest release.

**Comment by christiansakai** [comment]
Gotcha, thanks!

**Comment by davezuko** [comment]
FYI, still need to flesh out my replies, but since then:

Karma has been fixed (0.13.21)
Test warnings have been fixed (it's one of the latest commits)

**Comment by christiansakai** [comment]
Thank you so much!
